src/catch2/reporters/catch_reporter_helpers.cpp

```diff
@@ -235,4 +235,88 @@ namespace Catch {
         out << "\n\n" << std::flush;
     }
 
+    namespace {
+        struct SummaryColumn {
+            SummaryColumn( std::string _label, Colour::Code _colour ):
+                label( CATCH_MOVE( _label ) ), colour( _colour ) {}
+            SummaryColumn addRow( std::uint64_t count ) {
+                ReusableStringStream rss;
+                rss << count;
+                std::string row = rss.str();
+                for ( auto& oldRow : rows ) {
+                    while ( oldRow.size() < row.size() )
+                        oldRow = ' ' + oldRow;
+                    while ( oldRow.size() > row.size() )
+                        row = ' ' + row;
+                }
+                rows.push_back( row );
+                return *this;
+            }
+
+            std::string label;
+            Colour::Code colour;
+            std::vector<std::string> rows;
+        };
+
+        static void printSummaryRow( StringRef label,
+                                     std::vector<SummaryColumn> const& cols,
+                                     std::size_t row,
+                                     std::ostream& stream,
+                                     ColourImpl& colour ) {
+            for ( auto col : cols ) {
```

Comment:
Does this need to be a copy? From a quick glance I would say no, the code does not modify the `SummaryColumn` here.

(I know it is like this in the original code, but I am looking through it again here and hoo boy, it is a mess)